pkg/util/dynamichelper/dynamichelper_test.go

```diff
@@ -18,11 +20,89 @@ import (
 	kruntime "k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/rest/fake"
 
 	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
 	"github.com/Azure/ARO-RP/pkg/util/cmp"
 )
 
+type mockGVRResolver struct{}
+
+func (gvr mockGVRResolver) Refresh() error {
+	return nil
+}
+
+func (gvr mockGVRResolver) Resolve(groupKind, optionalVersion string) (*schema.GroupVersionResource, error) {
+	return &schema.GroupVersionResource{Group: "metal3.io", Version: "v1alpha1", Resource: "configmap"}, nil
+}
+
+func TestEsureDeleted(t *testing.T) {
+	ctx := context.Background()
+
+	for _, tt := range []struct {
+		uname   string
+		grpkind string
+	}{
+		{
+			uname:   "Test with the groupkind configmap",
+			grpkind: "configmap",
+		},
+		{
+			uname:   "Test with the groupkind baremetalhost",
+			grpkind: "baremetalhost.metal3.io",
+		},
```

Comment:
We have two test cases here with different group kinds, but `mockGVRResolver` only resolves one. So these two cases test the same code paths.

We should:
* Either get rid of test cases and subtests (`t.Run(...)`) and do a simple test with multiple sequential `EnsureDeleted` calls.
* Or have more meaningfull test cases & subtests. This will basically mean transforming sequential `EnsureDeleted` calls into sub tests.

Either option is fine.

Reply:
@m1kola - I have updated the PR based on your suggestions, please feel free to check and see if I understood you correctly. Thanks a lot.